### PR TITLE
Refactor alpine tests to use golden tables

### PR DESCRIPTION
**Pull request by scottsand-db**
- alpine tests now use golden tables
- changes to data reader as requested
- changes to log store tests as requested

### CODE REVIEW

**Review by zsxwing — APPROVED** [review]
LGTM! Left some minor comments

File: alpine/src/main/scala/io/delta/alpine/internal/DeltaLogImpl.scala [hunk 1]
```diff
@@ -31,14 +31,13 @@ private[internal] class DeltaLogImpl private(
...
-  with SnapshotManagement
-  with ReadOnlyLogStoreProvider {
+  with SnapshotManagement {
 
   override def getLogPath: Path = logPath
 
   override def getDataPath: Path = dataPath
 
-  lazy val store = createLogStore(hadoopConf)
+  lazy val store = ReadOnlyLogStore.createLogStore(hadoopConf)
```
[the remark this answers is not in the document]
Comment (scottsand-db) [reply alpine/src/main/scala/io/delta/alpine/internal/DeltaLogImpl.scala:40]: Sure. Just so you know, this `ReadOnlyLogStore` is the `alpine.internal.storage.ReadOnlyLogStore` object, not the exposed java abstract class

[the remark this answers is not in the document]
Comment (scottsand-db) [reply alpine/src/main/scala/io/delta/alpine/internal/DeltaLogImpl.scala:40]: But, ah. I forgot that you said this on the design doc:
> Yep, current we have only one implementation. It's not hard to add a LogStore API. We can hide LogStore right now and if someone asks, we can expose it and do a release quickly.

So, should I remove the `alpine.storage.ReadOnlyLogStore.java` and the 'alpine.internal.storage.ReadOnlyLogStore` object ?
Instead, just have a `alpine.storage.internal.ReadOnlyLogStore.scala` trait (not an object)

And then just have the `HDFSLogStore` implement that internal trait, and also refactor the log store suite?

File: alpine/src/main/scala/io/delta/alpine/internal/data/CloseableParquetDataIterator.scala [hunk 2]
```diff
@@ -20,16 +20,15 @@ import java.util.TimeZone
...
 
 private[internal] case class CloseableParquetDataIterator(
     dataFilePaths: Seq[String],
-    dataPath: String,
     schema: StructType,
     timeZoneId: String) extends CloseableIterator[RowParquetRecordJ] {
-  private val correctedTimeZoneId = if (null == timeZoneId) "UTC" else timeZoneId
-  private val readTimeZone = TimeZone.getTimeZone(correctedTimeZoneId)
+  private val readTimeZone = if (null == timeZoneId) TimeZone.getDefault
+    else TimeZone.getTimeZone(timeZoneId)
```
[the remark this answers is not in the document]
Comment (scottsand-db) [reply alpine/src/main/scala/io/delta/alpine/internal/data/CloseableParquetDataIterator.scala:31]: good to know. i'll start doing that going forwards

File: alpine/src/test/scala/io/delta/alpine/internal/DeltaTimeTravelSuite.scala [hunk 3]
```diff
@@ -17,143 +17,127 @@
...
+/**
+ * Instead of using Spark in this project to WRITE data and log files for tests, we have
+ * io.delta.golden.GoldenTables do it instead. During tests, we then refer by name to specific
+ * golden tables that that class is responsible for generating ahead of time. This allows us to
+ * focus on READING only so that we may fully decouple from Spark and not have it as a dependency.
+ *
+ * See io.delta.golden.GoldenTables for documentation on how to ensure that the needed files have
+ * been generated.
+ */
+class DeltaTimeTravelSuite extends FunSuite {
```
[the remark this answers is not in the document]
Comment (scottsand-db) [reply alpine/src/test/scala/io/delta/alpine/internal/DeltaTimeTravelSuite.scala:46]: good catch!

File: alpine/src/main/scala/io/delta/alpine/internal/DeltaLogImpl.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (scottsand-db) [reply alpine/src/main/scala/io/delta/alpine/internal/DeltaLogImpl.scala:40]: Just saw the Design Decision 5 > Option 3, so I'll go ahead and remove the public logstore API. thanks!

**Pull request merged**